src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs

```diff
@@ -70,15 +75,113 @@ public abstract class QueryableController : ApiController
 			}
 		}
 
+		protected async Task<IHttpActionResult> DeleteAsync(string application, string service, string collection,
+			ValueViewModel[] obj)
+		{
+			var serviceUri = GetServiceUri(application, service);
+			try
+			{
+
+				bool[][] results = new bool[obj.Length][];
+				for (int i = 0; i < obj.Length; i++)
+				{
+					//Serialize the key from the json body and put it into a string.
+					string keyquoted = JsonConvert.SerializeObject(obj[i].Key,
+						new JsonSerializerSettings {StringEscapeHandling = StringEscapeHandling.EscapeNonAscii});
+
+					//Fetch partition proxy.
+					var proxy = await GetServiceProxyForPartitionAsync<IQueryableService>(serviceUri, obj[i].PartitionId)
+						.ConfigureAwait(false);
+					//Perform delete opration.
+					results[i] = await Task.WhenAll(proxy.Select(p => p.DeleteAsync(collection, keyquoted)))
+						.ConfigureAwait(false);
+
+				}
+				return Ok(results);
```

Comment:
This should return something more descriptive. A mapping of keys to successes would be helpful. Alternately, I'm not sure this needs to delete more than 1 thing at a time.